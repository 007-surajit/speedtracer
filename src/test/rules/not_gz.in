hintlet_not_gz.js

# Input records 1 and 2
# Expected: HINTLET RECORD
# .html file
{"data":{"headers":{"Accept-Ranges":"bytes","Content-Length":"6436","Content-Type":"text/html","Date":"Fri, 04 Sep 1998 21:09:13 GMT","Last-Modified":"Sat, 09 Aug 2008 15:36:22 GMT","Server":"Apache/1.3.41 (Unix)"},"resourceId":"0NetworkResourceEvent31","responseCode":200,"url":"http://example.org/foo.html"},"time":5630.783999998122,"type":23}
{"data":{"contentLength":6436,"resourceId":"0NetworkResourceEvent31"},"time":5630.783999998122,"type":22}

# Input record 3 and 4
# Expected: HINTLET RECORD
# .css file
{"data":{"headers":{"Accept-Ranges":"bytes","Content-Length":"9875","Content-Type":"text/css","Date":"Fri, 04 Sep 1998 21:09:13 GMT","Last-Modified":"Sat, 09 Aug 2008 15:36:22 GMT","Server":"Apache/1.3.41 (Unix)"},"resourceId":"0NetworkResourceEvent31","responseCode":200,"url":"http://example.org/foo.css"},"time":5630.783999998122,"type":23}
{"data":{"contentLength":9875,"resourceId":"0NetworkResourceEvent31"},"time":5630.783999998122,"type":22}

# Input record 5 and 6
# Expected: HINTLET RECORD
# Javascript
{"data":{"headers":{"Accept-Ranges":"bytes","Content-Length":"9875","Content-Type":"text/javascript","Date":"Fri, 04 Sep 1998 21:09:13 GMT","Last-Modified":"Sat, 09 Aug 2008 15:36:22 GMT","Server":"Apache/1.3.41 (Unix)"},"resourceId":"0NetworkResourceEvent31","responseCode":200,"url":"http://example.org/foo.js"},"time":5630.783999998122,"type":23}
{"data":{"contentLength":9875,"resourceId":"0NetworkResourceEvent31"},"time":5630.783999998122,"type":22}

# Input record 7 and 8
# Expected: none
# image
{"data":{"headers":{"Accept-Ranges":"bytes","Content-Length":"9875","Content-Type":"image.png","Date":"Fri, 04 Sep 1998 21:09:13 GMT","Last-Modified":"Sat, 09 Aug 2008 15:36:22 GMT","Server":"Apache/1.3.41 (Unix)"},"resourceId":"0NetworkResourceEvent31","responseCode":200,"url":"http://example.org/foo.png"},"time":5630.783999998122,"type":23}
{"data":{"contentLength":9875,"resourceId":"0NetworkResourceEvent31"},"time":5630.783999998122,"type":22}

# Input record 9 and 10
# Expected: none
# too small
{"data":{"headers":{"Accept-Ranges":"bytes","Content-Length":"6","Content-Type":"text/html","Date":"Fri, 04 Sep 1998 21:09:13 GMT","Last-Modified":"Sat, 09 Aug 2008 15:36:22 GMT","Server":"Apache/1.3.41 (Unix)"},"resourceId":"0NetworkResourceEvent31","responseCode":200,"url":"http://example.org/foo.html"},"time":5630.783999998122,"type":23}
{"data":{"contentLength":6,"resourceId":"0NetworkResourceEvent31"},"time":5630.783999998122,"type":22}

# Input record 11 and 12
# Expected: none
# compressed with gzip
{"data":{"headers":{"Accept-Ranges":"bytes","Content-Length":"6436","Content-Type":"text/html","Content-Encoding":"gzip","Date":"Fri, 04 Sep 1998 21:09:13 GMT","Last-Modified":"Sat, 09 Aug 2008 15:36:22 GMT","Server":"Apache/1.3.41 (Unix)"},"resourceId":"0NetworkResourceEvent31","responseCode":200,"url":"http://example.org/foo.html"},"time":5630.783999998122,"type":23}
{"data":{"contentLength":6436,"resourceId":"0NetworkResourceEvent31"},"time":5630.783999998122,"type":22}

# Input record 13 and 14
# Expected: none
# compressed with bzip2
{"data":{"headers":{"Accept-Ranges":"bytes","Content-Length":"6436","Content-Type":"text/html","Content-Encoding":"bzip2","Date":"Fri, 04 Sep 1998 21:09:13 GMT","Last-Modified":"Sat, 09 Aug 2008 15:36:22 GMT","Server":"Apache/1.3.41 (Unix)"},"resourceId":"0NetworkResourceEvent31","responseCode":200,"url":"http://example.org/foo.html"},"time":5630.783999998122,"type":23}
{"data":{"contentLength":6436,"resourceId":"0NetworkResourceEvent31"},"time":5630.783999998122,"type":22}