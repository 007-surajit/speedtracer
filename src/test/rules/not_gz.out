{"hintletRule":"Uncompressed Resource","timestamp":5630.783999998122,"description":"URL http://example.org/foo.html was not compressed with gzip or bzip2","refRecord":1,"severity":3}
{"hintletRule":"Uncompressed Resource","timestamp":5630.783999998122,"description":"URL http://example.org/foo.css was not compressed with gzip or bzip2","refRecord":2,"severity":3}
{"hintletRule":"Uncompressed Resource","timestamp":5630.783999998122,"description":"URL http://example.org/foo.js was not compressed with gzip or bzip2","refRecord":3,"severity":3}
